sql/sql_base.cc

```diff
@@ -8583,20 +8583,32 @@ insert_fields(THD *thd, Name_resolution_context *context, const char *db_name,
                     tables->is_natural_join);
         DBUG_ASSERT(item->type() == Item::FIELD_ITEM);
         Item_field *fld= (Item_field*) item;
-        const char *field_db_name= field_iterator.get_db_name();
-        const char *field_table_name= field_iterator.get_table_name();
+        /*
+          TODO(cvicentiu) constructing these lex_cstrings could be avoided
+          if we updated all the users of field_iterator::get_xxx_name to
+          use LEX_CSTRING.
+        */
+        const LEX_CSTRING field_db_name= {
```

Comment:
btw, generally, please, prefer `Lex_table_name` (for table and db names) and `Lex_ident` (for other identifiers) to `LEX_CSTRING`. They're binary compatible, you can pass the former where the latter is expected, but they carry additional metadata for the compiler and can be compared easier.